fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/RangeSimplifier.java

```diff
@@ -0,0 +1,166 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.optimizer.rule.transformation.materialization;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.BoundType;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Range;
+import com.starrocks.sql.optimizer.Utils;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+public class RangeSimplifier {
```

Comment:
Give an example ?

Reply:
RangeSimplifier is used to judge whether range predicate can be rewritten.
for example, query predicate:  a < 10 and a > 1; view predicate: a < 100,
then RangeSimplifier will judge query predicate can be rewritten, and return the rewritten predicate: 1 < a < 10